src/main/java/org/folio/fqm/service/EntityTypeFlatteningService.java

```diff
@@ -221,21 +224,61 @@ private Pair<EntityTypeSource, List<EntityTypeColumn>> getConvertedSourceAndColu
   }
 
   private List<EntityTypeColumn> finalColumnConversion(EntityType flattenedEntityType) {
-    List<EntityTypeColumn> finalColumns = new ArrayList<>();
-    String toReplace = ":sourceAlias";
-    for (EntityTypeColumn column : flattenedEntityType.getColumns()) {
-      String sourceAlias = "\"" + column.getSourceAlias() + "\"";
-      String valueGetter = column.getValueGetter();
-      String filterValueGetter = column.getFilterValueGetter();
-      valueGetter = valueGetter.replace(toReplace, sourceAlias);
-      if (filterValueGetter != null) {
-        filterValueGetter = filterValueGetter.replace(toReplace, sourceAlias);
-      }
-      column.valueGetter(valueGetter);
-      column.filterValueGetter(filterValueGetter);
-      finalColumns.add(column);
+    return flattenedEntityType.getColumns()
+      .stream()
+      .map(column -> injectSourceAlias(column, column.getSourceAlias()))
```

Comment:
The sourceAlias gets pulled out here, instead of in `injectSourceAlias()` because that method takes a `T extends Field` parameter, and the sourceAlias comes from `EntityTypeColumn` objects, which is a subclass of `Field`. By using `T extends Field`, it lets `injectSourceAliasForObjectType()` recursively call `injectSourceAlias` with the properties from the object type

Reply:
Any reason not to just use Field?